util/golomb.cpp

```diff
@@ -137,8 +137,8 @@ struct binary_input_file_iterator
         return !operator==( other );
     }
 
-    T &                          operator*()        { return value; }
-    const T &                    operator*()  const { return value; }
+    reference                    operator*()        { return value; }
+    const reference              operator*()  const { return value; }
```

Comment:
GCC in /Wall gives a warning that the first const is ignored, since it is automatically a const reference as a result of the member function being const. It is still more clear to the human "compiler" this way IMO

Reply:
I don't get any warning when building with the makefile. The CPPFLAGS include -Wall.